src/Preg.php

```diff
@@ -30,16 +30,16 @@ final class Preg
      *
      * @throws PregException
      */
-    public static function match(string $pattern, string $subject, ?array &$matches = null, int $flags = 0, int $offset = 0): int
+    public static function match(string $pattern, string $subject, ?array &$matches = null, int $flags = 0, int $offset = 0): bool
```

Comment:
original idea was to mimic `preg_match` interface as much as possible

Reply:
But it does not make sense 🤷. In native `preg_match()` errors are not thrown, but side-stored and available via `preg_last_error()`, that's why it requires 3-state return type. In our case we throw exception if there's an error in the upstream, so we don't need `false` and can narrow it to boolean (instead of `0` / `1` from the upstream).